waddrmgr/manager.go

```diff
@@ -1470,6 +1503,238 @@ func (m *Manager) LastInternalAddress(account uint32) (ManagedAddress, error) {
 	return acctInfo.lastInternalAddr, nil
 }
 
+// ValidateAccountName validates the given account name and returns an error, if any.
+func ValidateAccountName(name string) error {
+	if name == "" {
+		str := "invalid account name, cannot be blank"
+		return managerError(ErrInvalidAccount, str, nil)
+	}
+	if name == "*" {
+		str := "invalid account name, cannot be '*'"
+		return managerError(ErrInvalidAccount, str, nil)
+	}
+	return nil
+}
+
+// NewAccount creates and returns a new account stored in the manager based
+// on the given account name.  If an account with the same name already exists,
+// ErrDuplicateAccount will be returned.  Since creating a new account requires
+// access to the cointype keys (from which extended account keys are derived),
+// it requires the manager to be unlocked.
+func (m *Manager) NewAccount(name string) (uint32, error) {
+	if m.watchingOnly {
+		return 0, managerError(ErrWatchingOnly, errWatchingOnly, nil)
+	}
+
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	if m.locked {
+		return 0, managerError(ErrLocked, errLocked, nil)
+	}
+
+	var account uint32
+	var coinTypePrivEnc []byte
+
+	// Validate account name
+	if err := ValidateAccountName(name); err != nil {
+		return account, err
+	}
+
+	// Check that account with the same name does not exist
+	_, err := m.lookupAccount(name)
+	if err == nil {
+		str := fmt.Sprintf("account with the same name already exists")
+		return account, managerError(ErrDuplicateAccount, str, err)
+	}
+
+	// Fetch latest account, and create a new account in the same transaction
+	err = m.namespace.Update(func(tx walletdb.Tx) error {
+		var err error
+		// Fetch the latest account number to generate the next account number
+		account, err = fetchLastAccount(tx)
+		if err != nil {
+			return err
+		}
+		account++
+		// Fetch the cointype key which will be used to derive the next account
+		// extended keys
+		_, coinTypePrivEnc, err = fetchCoinTypeKeys(tx)
+		if err != nil {
+			return err
+		}
+
+		// Decrypt the cointype key
+		serializedKeyPriv, err := m.cryptoKeyPriv.Decrypt(coinTypePrivEnc)
+		if err != nil {
+			str := fmt.Sprintf("failed to decrypt cointype serialized private key")
+			return managerError(ErrLocked, str, err)
+		}
+		coinTypeKeyPriv, err := hdkeychain.NewKeyFromString(string(serializedKeyPriv))
```

Comment:
The plain text coin type key bytes need to be zeroed as soon as they are used.
```Go
zero(serializedKeyPriv)
```

NOTE: This is intentionally before the error check so it happens regardless.